app/models/match.rb

```diff
@@ -11,22 +11,24 @@ class Match < ApplicationRecord
 
   validate :stage_xor_group
 
+  def evaluate_match_result
+    if match_scores.first == match_scores.second
```

Comment:
You're comparing objects here.. they will (probably) never be the same, therefore this method will return wrong results (a spec for this method should catch this)
You should use `match_scores.first/second.points` instead.

Reply:
As discusssed please remove this method from the PR and add it in the one adding a match finish method.